src/Paginator.php

```diff
@@ -861,10 +862,15 @@ protected function _getCacheId($page = null)
     // @codingStandardsIgnoreStart
     protected function _getCacheInternalId()
     {
+        $adapter            = $this->getAdapter();
+        $adapterToSerialize = $adapter instanceof DbSelect
+            ? $adapter->getArrayCopy()
+            : $adapter;
```

Comment:
My primary concern about this is that it introduces hard-coded logic for a single adapter. Maybe it might make more sense to have `DbSelect` implement `JsonSerializable` (and basically just rename `getArrayCopy()` to `jsonSerialize()`)? That way we can remove the hard-coded logic here.

Reply:
This was my original proposal but if we implement the interface `JsonSerializable` then another dependency to `ext-json` is made which is missing in the Composer configuration.
For the usage of the adapters the interface makes no sense because the `jsonSerialize` is used only internal and mark a method as internal which is implement by an interface looks strange.

One task for the next major version: extract the entire caching and rewriting the creation of a cache ID.

Reply:
Okay, so let's add a `@todo` above this code indicating that we ened to refactor for v2.

Reply:
nm, https://github.com/laminas/laminas-paginator/pull/6/files#diff-3739329a98f49b790bb2f2cddca7903fR156 contains that `@todo`.